crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -655,6 +682,53 @@ impl<'a, 'o> TimelineEventHandler<'a, 'o> {
         }
     }
 
+    /// Handle a beacon_info event and update the `BeaconInfoState`.
+    ///
+    /// TODO: Handle case of `m.beacon_info` event that is identical to the
+    /// first, except with live: false. This stops the live location share.
+    ///
+    /// TODO: How do you search for an existing beacon_info event in the
+    /// timeline by state key?
```

Comment:
If you have to do something like this, then the Timeline isn't the right place to do it; the timeline should eventually just be considered a "view" over events, not a model / source of truth in any ways, except when it comes to the view model itself.